src/services/completions.ts

```diff
@@ -2738,78 +2738,76 @@ export function getCompletionEntriesFromSymbols(
 
     function shouldIncludeSymbol(symbol: Symbol, symbolToSortTextMap: SymbolSortTextMap): boolean {
         let allFlags = symbol.flags;
-        if (!isSourceFile(location)) {
-            // export = /**/ here we want to get all meanings, so any symbol is ok
-            if (isExportAssignment(location.parent)) {
-                return true;
+        // export = /**/ here we want to get all meanings, so any symbol is ok

```

Comment:
All of this is simply outdented because I removed the wrapping `if` statement. Within the outdented codepaths I had to guard 2 places against missing `location.parent` and that's it.

cc @gabritto as you have a fresh context on all of this and related topics